build/ProjectsHelper.cs

```diff
@@ -0,0 +1,109 @@
+
+    private const string CoreProjectSelector = "OpenTelemetry.AutoInstrumentation*";
+    private const string TestApplicationSelector = "TestApplication.*";
+    private const string TestLibrarySelector = "TestLibrary.*";
+
+    public static IEnumerable<Project> GetManagedSrcProjects(this Solution solution)
+    {
+        return solution
+            .GetProjects(CoreProjectSelector)
+            .Where(x =>
+                // Should contain in the src directory
+                x.Directory.ToString().Contains(SrcDirName) &&
+                // Should not be native projects
+                !x.Name.Contains(NativeProjectMarker));
+    }
+
+    public static IEnumerable<Project> GetNativeSrcProjects(this Solution solution)
+    {
+        return solution
+            .GetProjects(CoreProjectSelector)
+            .Where(x =>
+                // Should contain in the src directory
+                x.Directory.ToString().Contains(SrcDirName) &&
+                // Should be native projects
+                x.Name.Contains(NativeProjectMarker));
+    }
+
+    public static IEnumerable<Project> GetManagedTestProjects(this Solution solution)
+    {
+        return solution.GetManagedUnitTestProjects()
+            .Concat(new[] { solution.GetManagedIntegrationTestProject() });
+    }
+
+    public static IEnumerable<Project> GetManagedUnitTestProjects(this Solution solution)
+    {
+        return solution
+            .GetProjects(CoreProjectSelector)
+            .Where(x =>
+                // Should contain in the test directory
+                x.Directory.ToString().Contains(TestDirName) &&
+                // Should not be native projects
+                !x.Name.Contains(NativeProjectMarker) &&
+                // Should be test projects
+                x.Name.EndsWith(TestsProjectMarker));
+    }
+
+    public static Project GetManagedIntegrationTestProject(this Solution solution)
+    {
+        return solution.GetProject(Projects.Tests.IntegrationTests);
+    }
+
+    public static IEnumerable<Project> GetTestApplications(this Solution solution)
+    {
+        var testApplications = solution.GetProjects(TestApplicationSelector);
+        var testLibraries = solution.GetProjects(TestLibrarySelector);
+
+        return testApplications.Concat(testLibraries);
+    }
+
+    public static Project GetTestMock(this Solution solution)
+    {
+        return solution.GetProject(Projects.Mocks.AutoInstrumentationMock);
+    }
+
+    public static IEnumerable<Project> GetWindowsOnlyTestApplications(this Solution solution)
+    {
+        return solution
+            .GetTestApplications()
+            .Where(x => x.Name.Contains("AspNet") || x.Name.EndsWith(NetFrameworkMarker));
+    }
+
+    public static IEnumerable<Project> GetCrossPlatformTestApplications(this Solution solution)
+    {
+        return solution
+            .GetTestApplications()
+            .Where(x => !x.Name.Contains("AspNet") && !x.Name.EndsWith(NetFrameworkMarker));
```

Comment:
I'll refactor "AspNet" later, this project should be renamed and reusing `NetFrameworkMarker`.